Update react to 18 and MUI to 5

timbrown5 opened the pull request:
- Update react version to 18 
- Update MUI to 5.10.10
- Replace react-split-pane with react-collapse-pane - to support react 18.
- Remove styling from TreeItems (this has changed for @mui 5).
-- I can look at adding and required styling back in - but most seemed irrelevant.

<!-- Please describe your pull request here. -->

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md  in your own repository 
-->

timbrown5 commented:
Seems functional:
![Screenshot 2022-10-19 at 15 07 38](https://user-images.githubusercontent.com/4447764/196717620-8465e230-19a8-403e-9923-75b90154058f.png)
![Screenshot 2022-10-19 at 15 07 58](https://user-images.githubusercontent.com/4447764/196717646-f1b1cc01-f5d8-4437-adc3-8fe5f309a18d.png)
![Screenshot 2022-10-19 at 15 08 28](https://user-images.githubusercontent.com/4447764/196717671-21768d39-a777-4b56-b1ba-882f9a234128.png)

The highlighting of TreeViewItems is new (and a bit ugly) - not sure where to disable it :/

timja commented:
nice work thanks!

you can generally apply css to the classes that they add if you can't find a specific option to turn off the issue with the TreeViewItems

timbrown5 commented:
I have fixed the highlighting and made the underlining of steps more obvious.

<img width="1483" alt="Screenshot 2022-10-20 at 11 17 11" src="https://user-images.githubusercontent.com/4447764/196922629-378e55a3-0ea2-4c67-bb74-094cb79b73e3.png">

Here Build is selected and I'm hovering over mvn. I think it's a little too subtle, but I can't find an acceptable mix of Jenkins vars to use to make the difference clearer.

Here it is in
<img width="1464" alt="Screenshot 2022-10-20 at 11 11 04" src="https://user-images.githubusercontent.com/4447764/196922043-7f94b395-bac9-4eac-8f21-2f383179bf4c.png">
 dark mode:

Material Red:
<img width="1472" alt="Screenshot 2022-10-20 at 11 10 28" src="https://user-images.githubusercontent.com/4447764/196922724-72ef58fb-1964-419a-8c78-23726deb73e2.png">

Material Green:
<img width="1479" alt="Screenshot 2022-10-20 at 11 12 58" src="https://user-images.githubusercontent.com/4447764/196922000-e9e14678-fbd6-48e4-9d0d-4a19320be461.png">

timbrown5 commented:
Fixed the weird formatting in the dark theme image above:
<img width="1491" alt="Screenshot 2022-10-20 at 11 22 47" src="https://user-images.githubusercontent.com/4447764/196924364-df4b1a15-5ac5-44a3-b5e9-c7b9e9a5d865.png">

Also removed some unneeded divs (which added extra padding);
<img width="1479" alt="Screenshot 2022-10-20 at 11 23 27" src="https://user-images.githubusercontent.com/4447764/196924483-212874b5-5599-4b90-b336-dac42ea323c3.png">

<img width="1478" alt="Screenshot 2022-10-20 at 11 24 16" src="https://user-images.githubusercontent.com/4447764/196924518-7f2a5fb2-ecea-4fb9-b5f7-a83b0adc095e.png">

timbrown5 commented:
Just to check, is there anything you are waiting on me for?

timja commented:
No just time to check it

timja commented on src/main/frontend/pipeline-console-view/pipeline-console/main/PipelineConsole.tsx:
```diff
@@ -380,35 +371,36 @@ export class PipelineConsole extends React.Component<
+            <DataTreeView
+              onNodeSelect={this.handleActionNodeSelect}
+              onNodeToggle={this.handleToggle}
+              selected={this.state.selected}
```
Active state doesn't seem as clear as before

Before
<img width="316" alt="image" src="https://user-images.githubusercontent.com/21194782/197337916-9365dfdb-8b1e-47f5-9588-defb9f0ff39a.png">


After
<img width="352" alt="image" src="https://user-images.githubusercontent.com/21194782/197337906-a3e26759-f2d4-409b-bd06-e37872bfb741.png">

timja reviewed (commented):
Looking great 2 minor issues

timbrown5 commented:
Sorry, didn't mean to pressure you. Thanks for having a look. Those issues should be solved now (along with the merge conflict).

timbrown5 commented:
Sorry, just notice that the focus isn't working on my lastest build. I will have a look.

timja commented:
> Sorry, didn't mean to pressure you. Thanks for having a look. Those issues should be solved now (along with the merge conflict).

was not pressured, feel free to ping me to remind me don't feel bad :)

timbrown5 commented:
Just a thought, have I broken the Graph View tests? I remember removing the test framework (it's incompatible with react 18) but haven't done anything to get them working again (but the build doesn't error).

timbrown5 commented:
Fixed an issue when running tests in VSCode UI, but not sure how to check they run as part of the build.

timja commented:
> Fixed an issue when running tests in VSCode UI, but not sure how to check they run as part of the build.

`npm run test` should do it, and `npm run mvntest` should be run as part of the build
https://github.com/jenkinsci/pipeline-graph-view-plugin/blob/92eda332ca39bd333f5601f99ac3a3e981558b5e/package.json#L8

timbrown5 commented:
I get some pretty-check warnings when I run that, but no test output - just
a little nervous the tests are disabled.
I will try breaking one of the tests (locally) to check that npm test is
working.

On Mon, 24 Oct 2022, 12:04 Tim Jacomb, ***@***.***> wrote:

> Fixed an issue when running tests in VSCode UI, but not sure how to check
> they run as part of the build.
>
> npm run test should do it, and npm run mvntest should be run as part of
> the build
>
> https://github.com/jenkinsci/pipeline-graph-view-plugin/blob/92eda332ca39bd333f5601f99ac3a3e981558b5e/package.json#L8
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/161#issuecomment-1288865885>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABB54FBQ6BSAVEYGJSQKANTWEZULVANCNFSM6AAAAAARJFO34M>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

timbrown5 commented:
I think the tests weren't running locally as `npm run prettier-check` exited with 1 due to warnings.
Now I get:
```
npm run test

> pipeline-graph-view-plugin@1.0.0 test
> npm run jest prettier-check

npm ERR! Missing script: "jest"
npm ERR! 
npm ERR! Did you mean this?
npm ERR!     npm test # Test a package
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
```

Running jest directly works:
```
$  ./node_modules/.bin/jest 
 PASS  src/main/frontend/pipeline-graph-view/pipeline-graph/main/PipelineGraphLayout.spec.ts
  PipelineGraphLayout
    createNodeColumns
      ✓ returns no columns if no stages (2 ms)
      ✓ returns proper columns (unstableSmokes) (2 ms)
      ✓ returns proper columns (complexSmokes) (1 ms)
      ✓ returns proper columns (GH#63.1) (1 ms)
      ✓ returns proper columns (GH#63.2) (1 ms)
      ✓ returns proper columns (GH#50) (1 ms)
      ✓ returns proper columns (GH#18) (1 ms)

Test Suites: 1 passed, 1 total
Tests:       7 passed, 7 total
Snapshots:   0 total
Time:        3.054 s, estimated 4 s
Ran all test suites.
```

timbrown5 commented:
(Converted to draft to prevent this being landed broken - as I have deliberately broken a test to check CI runs npm tests).

timja commented on package.json:
```diff
@@ -23,36 +23,41 @@
-    "react": "16.14.0",
-    "react-dom": "16.14.0",
-    "react-split-pane": "0.1.92"
+    "jest-cli": "^29.2.1",
```
should be in dev dependencies

timbrown5 commented:
Test running locally and in CI:
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fpipeline-graph-view-plugin/detail/PR-161/12/pipeline/28#step-35-log-274

timja commented on src/main/frontend/pipeline-console-view/pipeline-console/main/Ansi.tsx:
```diff
@@ -148,19 +148,28 @@ export function tokenizeANSIString(input?: string): string[] | Result[] {
+        result.push(
+          <div
+            dangerouslySetInnerHTML={{ __html: codeOrString }}
+            key={`${key}-${i}`}
```
Still getting unique key issues on some steps:

![image](https://user-images.githubusercontent.com/21194782/197717730-828faf37-4671-4c98-a1e9-0f17f717aa06.png)

Build -> mvn

on this pipeline: https://github.com/timja-org/junit-attachments-test/tree/master

timbrown5 commented:
> not sure what was going here, with my changes it definitely works locally =/

Works locally for me now too 🤷.

timbrown5 commented:
I wonder if the change from `mvn run test` to `mvn test` manages to find the local `jest` executable rather than needing it to be in PATH (which is what I seem to find before)?
